pkgs/native_assets_cli/lib/src/api/c_code_asset.dart

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'asset.dart';
+
+/// A code [Asset] which respects the C application binary interface (ABI).
```

Comment:
(Nit: An ABI is not language specific, that's why it's called Binary. Linux and Windows have different x64 ABIs, and C compiler compiles to the *platform's* ABI, but so does any other native compiler. So it's not a/the C ABI, just an ABI, likely the platform's default ABI.)

Reply:
So, should this the be a `NativeCodeAsset`? I believe that to be confusing, because Java/Kotlin and Swift/ObjectiveC are also labeled as 'native' to Android and iOS respectively.

Reply:
I don't know if it needs to be renamed. 

Are there any other kinds of code asset? If so, what are they named?
If not, can this just be `CodeAsset`?
If we expect other code assets, and those are also "native" (because *all* code assets are "native" in some sense), then we can't use "Native" here. But that is what I'd use otherwise. It's *machine code* for the platform CPU, which is also assumed to follow the platform's default ABI conventions (at least calling convention and stack convention, and if it calls into the platform kernel, then also those ABIs).

My main worry is that the phrase "C ABI" isn't well-defined, and therefore is not a good *description*.

If it just means [`cdecl`](https://en.wikipedia.org/wiki/X86_calling_conventions), then say that. If it means "platform standard calling convention" then say that. If it means something different, we should figure out *precisely* what that is, so we can write it.

Reply:
Okay, we're going for `NativeCodeAsset`, and say that calling Java/Kotlin/Swift "native" is wrong.